### ISSUE TITLE
Rebuilding webassembly library produces empty output until cache is deleted

### ISSUE THREAD

**Issue by Pyrolistical** [ziglang/zig#12864]
### Zig Version

0.9.1

### Steps to Reproduce

1. Remove `zig-cache`
2. Build a webassembly library
3. Record base64 wasm output
4. Re-run build
5. See base64 wasm output is now different

### Expected Behavior

```
$ cat test.zig
export fn foo() void {}

$ rm -rf zig-cache

; initial build is correct
$ zig build-lib ./test.zig -target wasm32-freestanding -dynamic && < test.wasm | base64`      
AGFzbQEAAAABBAFgAAADAgEABAUBcAEBAQUDAQABBggBfwFBgIAECwcQAgZtZW1vcnkCAANmb28AAAoFAQMADws=

; re-build produces identical result
$ zig build-lib ./test.zig -target wasm32-freestanding -dynamic && < test.wasm | base64
AGFzbQEAAAABBAFgAAADAgEABAUBcAEBAQUDAQABBggBfwFBgIAECwcQAgZtZW1vcnkCAANmb28AAAoFAQMADws=
```

### Actual Behavior

```
$ cat test.zig
export fn foo() void {}

$ rm -rf zig-cache

; initial build is correct
$ zig build-lib ./test.zig -target wasm32-freestanding -dynamic && < test.wasm | base64`      
AGFzbQEAAAABBAFgAAADAgEABAUBcAEBAQUDAQABBggBfwFBgIAECwcQAgZtZW1vcnkCAANmb28AAAoFAQMADws=

; nothing changed, but re-building produces different result
$ zig build-lib ./test.zig -target wasm32-freestanding -dynamic && < test.wasm | base64
AGFzbQEAAAAEBQFwAQEBBQMBAAEGCAF/AUGAgAQLBwoBBm1lbW9yeQIA
```

If the `zig-cache` is deleted, a correct build will be produced once again.

`AGFzbQEAAAABBAFgAAADAgEABAUBcAEBAQUDAQABBggBfwFBgIAECwcQAgZtZW1vcnkCAANmb28AAAoFAQMADws=` disassembled is
```
(module
  (table $table0 1 1 funcref)
  (memory $memory (;0;) (export "memory") 1)
  (global $global0 (mut i32) (i32.const 65536))
  (func $foo (;0;) (export "foo")
    return
  )
)
```

`AGFzbQEAAAAEBQFwAQEBBQMBAAEGCAF/AUGAgAQLBwoBBm1lbW9yeQIA` disassembled is
```
(module
  (table $table0 1 1 funcref)
  (memory $memory (;0;) (export "memory") 1)
  (global $global0 (mut i32) (i32.const 65536))
)
```
Note the missing `foo` export.

**Comment by Pyrolistical** [comment]
Not a problem on `0.10.0-dev.4060+61aaef0b0`. Don't know if project cares to fix it in 0.9.1

**Comment by Luukdegram** [comment]
The issue tracker is for master only. Fixes are not backported to previous releases. Thanks for the report and verifying it was fixed in master. Closing this issue.
